src/basic_recipes/datashader.jl

```diff
@@ -0,0 +1,388 @@
+            end
+            # update the value in out2 directly in this loop
+            final_value = local_op(value(op, val))
+            if isfinite(final_value)
+                mini = min(final_value, mini)
+                maxi = max(final_value, maxi)
+            end
+            @inbounds out2[i, j] = final_value
+        end
+    end
+    return (mini, maxi)
+end
+
+
+export AggAny, AggCount, AggMean, AggSum, AggSerial, AggThreads
+
+end
+
+using ..PixelAggregation
+
+function equalize_histogram(matrix; nbins=256 * 256)
+    h_eq = StatsBase.fit(StatsBase.Histogram, vec(matrix); nbins=nbins)
+    h_eq = normalize(h_eq; mode=:density)
+    cdf = cumsum(h_eq.weights)
+    cdf = cdf / cdf[end]
+    edg = h_eq.edges[1]
+    # TODO is this the correct linear interpolation?
+    return Makie.interpolated_getindex.((cdf,), matrix, (Vec2f(first(edg), last(edg)),))
+end
+
+"""
+    datashader(points::AbstractVector{<: Point})
+
+Points can be any array type supporting iteration & getindex, including memory mapped arrays.
+If you have x + y coordinates seperated and want to avoid conversion + copy, consider using:
+```Julia
+using Makie.StructArrays
+points = StructArray{Point2f}((x, y))
+datashader(points)
+```
+Do pay attention though, that if x and y don't have a fast iteration/getindex implemented, this might be slower then just copying it into a new array.
+
+For best performance, use `method=Makie.AggThreads()` and make sure to start julia with `julia -t8` or have the environment variable `JULIA_NUM_THREADS` set to the number of cores you have.
+
+## Attributes
+
+### Specific to `DataShader`
+
+- `agg = AggCount()` can be `AggCount()`, `AggAny()` or `AggMean()`. User extendable by overloading:
+
+
+    ```Julia
+        struct MyAgg{T} <: Makie.AggOp end
+        MyAgg() = MyAgg{Float64}()
+        Makie.PixelAggregation.null(::MyAgg{T}) where {T} = zero(T)
+        Makie.PixelAggregation.embed(::MyAgg{T}, x) where {T} = convert(T, x)
+        Makie.PixelAggregation.merge(::MyAgg{T}, x::T, y::T) where {T} = x + y
+        Makie.PixelAggregation.value(::MyAgg{T}, x::T) where {T} = x
+    ```
+
+- `method = AggThreads()` can be `AggThreads()` or `AggSerial()`.
+- `async_latest::Bool = true` will calculate aggregation in a task, and skip any zoom/pan updates while busy. Great for interaction, but must be disabled for saving to e.g. png or when inlining in documenter.
+
+- `global_post::Function = Makie.equalize_histogram` function which gets called on the whole aggregation array before display (`global_post(final_aggregation_result)`).
+- `local_post::Function = identity` function which gets call on each element after aggregation (`map!(x-> local_post(x), final_aggregation_result)`).
+
+- `point_func::Function = identity` function which gets applied to every point before aggregating it.
+- `binfactor::Number = 1` factor defining how many bins one wants per screen pixel. Set to n > 1 if you want a corser image.
```

Comment:
I named it `binfactor` but I'm not sure if I like the name. Maybe `binsize` is more accurate?

Also "coarser"